crates/neon/src/types_impl/extract/with.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{
+    context::{Context, Cx},
+    result::JsResult,
+    types::extract::TryIntoJs,
+};
+
+/// Wraps a closure that will be lazily evaluated when [`TryIntoJs::try_into_js`] is
+/// called.
+///
+/// Useful for executing arbitrary code on the main thread before returning from a
+/// function exported with [`neon::export`](crate::export).
+///
+/// ## Example
+///
+/// ```
+/// # use neon::{prelude::*, types::extract::{TryIntoJs, With}};
+/// use std::time::Instant;
+///
+/// #[neon::export(task)]
+/// fn sum(nums: Vec<f64>) -> impl for<'cx> TryIntoJs<'cx> {
+///     let start = Instant::now();
+///     let sum = nums.into_iter().sum::<f64>();
+///     let log = format!("sum took {} ms", start.elapsed().as_millis());
+///
+///     With(move |cx| -> NeonResult<_> {
+///          cx.global::<JsObject>("console")?
+///             .call_method_with(cx, "log")?
+///             .arg(cx.string(log))
+///             .exec(cx)?;
+///
+///         Ok(sum)
```

Comment:
Is the indentation off here?

Reply:
Good eye! The `cx.global` line was off by exactly one space.